code/modules/mob/living/carbon/xenomorph/xenoattacks.dm

```diff
@@ -83,7 +83,8 @@
 	switch(X.a_intent)
 		if(INTENT_HELP)
 			if(on_fire)
-				fire_stacks = max(fire_stacks - 1, 0)
+				while(fire_stacks && do_after(X, 5, TRUE, src, BUSY_ICON_FRIENDLY))
```

Comment:
it means that the first stack removed will happened 0,5 sec after you first click, rather than being instantanious now.

Is this what you want?